platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/synchronization/TeachingSynchronizer.java

```diff
@@ -95,13 +102,16 @@ public class TeachingSynchronizer {
      * 		if there is a thread stuck on a blocking IO
      * 		operation that will never finish due to a
      * 		failure.
+     * @param requestToStopTeaching
+     *      a function to check periodically if teaching should be stopped
      */
     public TeachingSynchronizer(
             final ThreadManager threadManager,
             final MerkleDataInputStream in,
             final MerkleDataOutputStream out,
             final MerkleNode root,
-            final Runnable breakConnection) {
+            final Runnable breakConnection,
```

Comment:
null annotations and checks are missing

Reply:
Both the field and the arg are annotated as `@Nullable`